src/lib/validate-input.js

```diff
@@ -0,0 +1,412 @@
+'use strict';
+
+import {sizeOfShape, Tensor} from './tensor.js';
+
+/**
+ * Check the tensor whether it is a 1-D tensor and its length is equal to `expectedSize`.
+ * @param {Tensor} a
+ * @param {Number} expectedSize
+ * @param {String} name
+ */
+function check1DTensorWithSize(a, expectedSize, name) {
+  if (a) {
+    if (a.rank !== 1) {
+      throw new Error(`The parameter ${name} is not a 1-D tensor.`);
+    } else {
+      if (a.shape[0] !== expectedSize) {
+        throw new Error(`The length ${a.shape[0]} of the ${name} values is not equal to the ` +
+          `size ${expectedSize} of the input dimension denoted by options.axis.`);
+      }
+    }
+  }
+}
+
+export function validateInput(op, args) {
```

Comment:
Should we have specific method (or even separate file) for each op's validation? Say, `validateConv2d` etc.,

Reply:
if we keep input validation, and if we keep it separate, separate methods are probably indeed much cleaner; I'm not sure separate files are necessary warranted, but I don't mind strongly either

Reply:
Regarding to [discussion](https://github.com/webmachinelearning/webnn-baseline/pull/1#issuecomment-1031184302), it looks like we agree to keep the input validation code. So could you please make separate methods for them? I agree with you we can skip the separate files.